### DISCUSSION TITLE
Possibility to change date format DateFilter

### DISCUSSION THREAD

**Discussion by lucassmacedo** [rappasoft/laravel-livewire-tables#1073] [Feature Requests]
It would be very interesting a way to change the date format to other standards such as Brazilian

```

        if (DateTime::createFromFormat('d/m/Y', $value) === false) {
            return false;
        }
```

**Comment by lrljoe** [comment]
Good idea, I'm actually in the code for date filters at the mo so I'll stick a PR in with a default of current, but option to change this.
I'll need to write the tests etc so might be a day or two before the PR is opened

**Comment by lrljoe** [comment]
This is still on my to-do list.

However, I have stuck together a package of add-on filters very quickly which includes a Flatpickr Date Picker that **does** support custom date formats
[See here for details](https://github.com/LowerRockLabs/LaravelLivewireTablesAdvancedFilters), you can either take what I wrote and incorporate it manually, or I am planning to keep it maintained, and eventually PR it into core once its been used/abused suitably.

**Comment by lrljoe** [comment]
I apologise!  This fell off my radar yet again, but do feel free to use the add-on filter in the interim.

**Comment by lrljoe** [comment]
Just have some tests to write, so won't be in the next release, but expect this in the one that follows.

**Comment by lrljoe** [comment]
This will be in 2.13
2.12 should be released shortly, and 2.13 won't be anywhere near as many updates.  
I imagine 2.13 will be released later this month.

**Comment by lrljoe** [comment]
I tell a lie, 2.15, there were a couple of interim items.

**Comment by lrljoe** [comment]
I've been reviewing this over and over.

I fully appreciate that various cultures and nations use different date formats.  But this relates to how it is stored in the database, which I'd assume you're using Y-m-d for.

If this is really in relation to how it's presented once it's applied in the pills/toolbar area, then the functionality already exists for customising this.

The date picker itself is a browser native one, so it'll show the format however the locale is configured in browser

If I've missed something, do let me know!  There are some developments for filters in the works however

**Comment by lrljoe** [comment]
I'm going to add this into version 3.0

**Comment by lrljoe** [comment]
@lucassmacedo - After a LOT of looking into this, the key issue at the moment is that the browser itself expects Y-m-d format for the date picker in order to actually function.  If you present something else, then the native browser date picker won't work as you'd expect.

The v3 DateRange filter does support varied formats, and I'll be adding a standard DatePicker which will also support date formats.

I have however added customisable filter pills formats into V3 as an immediate feature.
